plugins/javatemplates/src/main/java/org/apache/struts2/views/java/simple/TextFieldHandler.java

```diff
@@ -31,8 +31,8 @@
     public void generate() throws IOException {
         Map<String, Object> params = context.getParameters();
         Attributes attr = new Attributes();
-
-        attr.add("type", "text")
+        Object type = params.get("type");
+        attr.add("type", type == null ? "text" : (String) type)
```

Comment:
Maybe you could refactor the rest as well and use `String.valueOf` or leave it as is, up to you :)